### PR TITLE
💪  Making a difference between write and read types.

### PR DESCRIPTION
**Pull request by nicolasgramlich**
- [x] Code
- [x] Tests
- [x] Docs

### CODE REVIEW

**Comment by nicolasgramlich** [comment]
🦆 Of course it's fine in IntelliJ, but not on the command line... investigating 🦆

**Comment by SylvainEstevez** [comment]
Given the added complexity, a quick doc about the generics would be welcome :)

File: src/classes/sequelize-service.ts [hunk 1]
```diff
@@ -40,47 +40,47 @@ import { ICreateSession } from '../interfaces/create-session';
 import { IDeleteSession } from '../interfaces/delete-session';
 
 @injectable()
-export class SequelizeService<A, CP extends {} = {}> extends Service implements ISequelizeService<A, CP> {
-  private primaryKeyField: keyof A = 'id' as keyof A;
-  protected computedPropertiesManager: IComputedPropertiesManager<A, CP>;
+export class SequelizeService<W, R extends W, C extends {} = {}> extends Service implements ISequelizeService<W, R, C> {
```
Comment (SylvainEstevez) [inline src/classes/sequelize-service.ts]: Should `W` extend `{}` ?

Comment (nicolasgramlich) [reply src/classes/sequelize-service.ts]: Same for `C`?

Comment (nicolasgramlich) [reply src/classes/sequelize-service.ts]: See: https://github.com/bluebirds-blue-jay/sequelize-service/commit/76123161a6908b18f7644ae6dfd1dccedc8943a8

**Comment by nicolasgramlich** [comment]
@SylvainEstevez: Everything is address. We can merge this, followed by with some followup improvements #5

File: src/types/values.ts [hunk 2]
```diff
@@ -1 +1 @@
-export type TValues<A> = { [key in keyof Partial<A>]: A[key] };
\ No newline at end of file
+export type TValues<T> = Partial<T>;
```
Comment (SylvainEstevez) [inline src/types/values.ts]: :)

**Pull request merged**